controllers/pullRequestsController.js

```diff
@@ -1,57 +1,82 @@
 const githubService = require('../services/githubService')
 
+/**
+ * Get Latest PRs in open or stale state for Real Dev Squad repos
+ *
+ * @param {boolean} page page number to retrieve
+ * @param {boolean} perPage number of PRs per page
+ * @param {boolean} isOpen boolean value indication open or stale
+ * @param {string} username retrieve PRs of username
+ *
+ * @returns {Array} List of PRs based on provided conditions
+ */
+const getPRs = async ({ page = 1, perPage = 10, isOpen = true, username = null }) => {
+  try {
+    const { data } = await githubService.fetchPRs({ page, perPage, isOpen, username })
+    if (!data.total_count) {
+      return []
+    }
+    return githubService.extractPRdetails(data)
+  } catch (err) {
+    logger.error(`Error while processing pull requests: ${err}`)
+    throw err
+  }
+}
+
 /**
  * Collects all pull requests and sends only required data for each pull request
  *
  * @param req {Object} - Express request object
  * @param res {Object} - Express response object
  */
-
 const getUserPRs = async (req, res) => {
   try {
-    const { data } = await githubService.fetchPRsByUser(req.params.username)
+    const { username } = req.params
+
+    let message
+    const prs = await getPRs({ username })
 
-    if (data.total_count) {
-      const allPRs = githubService.extractPRdetails(data)
-      return res.json({
-        message: 'Pull requests returned successfully!',
-        pullRequests: allPRs
-      })
+    if (prs.length) {
+      message = 'User PRs'
+    } else {
+      message = 'No pull requests found!'
     }
+
     return res.json({
-      message: 'No pull requests found!',
-      pullRequests: []
+      message: message,
+      pullRequests: prs
     })
   } catch (err) {
-    logger.error(`Error while processing pull requests: ${err}`)
     return res.boom.badImplementation('Something went wrong please contact admin')
   }
 }
 
 /**
- * Get stale PRs in open state for Real Dev Squad repos
+ * Get Latest PRs in stale state for Real Dev Squad repos
```

Comment:
We return `oldest stale PRs` first
Check here: https://api.realdevsquad.com/pullrequests/stale

Reply:
Latest stale PRs would become same as `pullrequests/open`

Reply:
Oops, forgot to update the comment here ig

Reply:
Latest PR in stale state means that latest stale PRs first right?